src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | object {
```

Comment:
The return type `GroupsMap<T> | object` can be simplified to just `GroupsMap<T>` since an empty object `{}` is also a valid `GroupsMap<T>`. This simplification will make the function's return type more specific and easier to understand.